source/ncengine/graphics2/diligent/resource/SinkIndexBufferResource.cpp

```diff
@@ -21,28 +21,35 @@ SinkIndexBufferResource::SinkIndexBufferResource(Diligent::IDeviceContext& conte
 
 void SinkIndexBufferResource::Update(Diligent::IDeviceContext& context,
                                      std::span<const uint32_t> colorSources,
-                                     std::span<const uint32_t> depthSources)
+                                     std::span<const uint32_t> depthSources,
+                                     bool hasPostProcessSource)
 {
     NC_ASSERT(colorSources.size() <= 4u, "Only four color sources supported.");
-    NC_ASSERT(depthSources.size() <= 4u, "Only four depth sources supported.");
+    NC_ASSERT(depthSources.size() <= 3u, "Only three depth sources supported.");
 
-    auto colorSourcesArray = std::array<uint32_t, 4u>{0u};
-    auto depthSourcesArray = std::array<uint32_t, 4u>{0u};
+    auto colorSourcesArray = std::array<int32_t, 4u>{-1};
+    auto depthSourcesArray = std::array<int32_t, 3u>{-1};
 
     for (auto i = 0u; i < colorSources.size(); i++)
     {
-        colorSourcesArray.at(i) = colorSources[i];
+        if (colorSources[i] != std::numeric_limits<uint32_t>::max())
```

Comment:
It was like this, but writing to the array based on the span size isn't safe, especially with `[]`. I don't know what the source object is, but span has a second template argument to declare a static size. Or you could just assert sizes are equal.

Constants would be nice here too.

Not sure on this one, but wasn't there another index type for max() - 1? Does that not come into play here? If it does, it would overflow the int, otherwise ignore me.

Reply:
I think we're safe here re: span not having a size. We default initialize the array, so we don't have to worry about  the span being less than the array size, because those values will just remain the defaults.  And,  we already have a static assert that the span size must be equal to or less than the array size:

```
    NC_ASSERT(colorSources.size() <= 4u, "Only four color sources supported.");
    NC_ASSERT(depthSources.size() <= 3u, "Only three depth sources supported.");
```
LMK if I'm missing something though.

Re: Checking for max-1: I can't remember my reasoning on omitting this check, so I've added it back in.